astropy/coordinates/baseframe.py

```diff
@@ -1273,8 +1308,8 @@ def is_equivalent_frame(self, other):
         """
         if self.__class__ == other.__class__:
             for frame_attr_name in self.get_frame_attr_names():
-                if np.any(getattr(self, frame_attr_name) !=
-                          getattr(other, frame_attr_name)):
+                if not self._frameattr_equiv(getattr(self, frame_attr_name),
```

Comment:
I see that this is keeping the same behaviour as previously which makes me wonder. Was there a good reason not to return ```False``` if ```self``` or ```other``` has frame attributes that are not on the other frame?

For example, perhaps we should add:

```python
if set(self.get_frame_attr.names().keys()) != set(other.get_frame_attr_names().keys()):
   return False
```

Reply:
Isn't it already guaranteed that those two sets of names are the same given that `self` and `other` are the same class?

Reply:
I agree with @mhvk here, *but* I think @StuartLittlefair is right that this should be a fix in #8205